packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -407,6 +412,15 @@ class MethodChannel {
       return codec.encodeErrorEnvelope(code: 'error', message: e.toString(), details: null);
     }
   }
+
+  /// Returns the number of invoke methods waiting for response.
+  ///
+  /// It currently counts the number of pending [invokeMethod] calls.
+  ///
+  /// This method is useful for a testing framework like Espresso to monitor
+  /// ongoing platform channel messages and decide whether the app is idle.
+  static int get pendingChannelInvokeMethodCount => _pendingChannelInvokeMethodCount;
```

Comment:
I think if this were exposed through a listener registration similar to `SchedulerBinding.instance.addPostFrameCallback()`, it would clean up some smells elsewhere in this PR, such as the 10ms polling.

It'd probably amount to adding API to `SchedulerBinding` - something like `addMethodChannelsIdleCallback()` (and moving the state variable into the binding, which also means it no longer needs to be static, which is nice)

Then again, this work isn't being _scheduled_  in the way that fits perfectly into the existing `SchedulerBinding` model. @Hixie @goderbauer wdyt?